orc/orcarm.c

```diff
@@ -64,6 +64,29 @@ orc_arm_reg_name (int reg)
   return gp_regs[reg&0xf];
 }
 
+const char *
+orc_arm64_reg_name (int reg, int reg_bits)
+{
+  static const char *gp_regs64[] = {
+     "x0",  "x1",  "x2",  "x3",  "x4",  "x5",  "x6",  "x7",  "x8",  "x9",
+    "x10", "x11", "x12", "x13", "x14", "x15", "x16", "x17", "x18", "x19",
+    "x20", "x21", "x22", "x23", "x24", "x25", "x26", "x27", "x28", "x29",
+    "x30",  "sp"
+  };
+  static const char *gp_regs32[] = {
+     "w0",  "w1",  "w2",  "w3",  "w4",  "w5",  "w6",  "w7",  "w8",  "w9",
+    "w10", "w11", "w12", "w13", "w14", "w15", "w16", "w17", "w18", "w19",
+    "w20", "w21", "w22", "w23", "w24", "w25", "w26", "w27", "w28", "w29",
+    "w30",  "sp"
+  };
```

Comment:
Don't you need to add "ERROR" to mark padding between registers?

Reply:
I think the below lines may handle this case?

Reply:
No, I'm talking about the space between the valid registers.

Reply:
E.g., (from orcrules-neon.c)
```
const char *orc_neon_reg_name_quad (int reg)
{
  static const char *vec_regs[] = {
    "q0", "ERROR", "q1", "ERROR",
    "q2", "ERROR", "q3", "ERROR",
    "q4", "ERROR", "q5", "ERROR",
    "q6", "ERROR", "q7", "ERROR",
    "q8", "ERROR", "q9", "ERROR",
    "q10", "ERROR", "q11", "ERROR",
    "q12", "ERROR", "q13", "ERROR",
    "q14", "ERROR", "q15", "ERROR",
  };

  if (reg < ORC_VEC_REG_BASE || reg >= ORC_VEC_REG_BASE+32) {
    return "ERROR";
  }
```

Reply:
Yes, AArch32 use 32 64-bit vector registers. To be used as 128-bit registers, two registers are merged. So, this array represents 16 128-bit vector registers.

`orc_arm64_reg_name` in this PR is for generic registers, not vector registers.
Even for the vector registers (which is updated later), they are already 128-bit in AArch64 (128-bit is maximum width). So, "ERROR" padding will be not required.

Reply:
oh... Sorry, I just understand your concerns. But, it's no problem.
Unlike Arm32, AArch64 does not use register merge. It meant that the first halves of registers are used to represent narrower registers.

So, AArch64 has 32 64-bit generic registers (X) and their first halves are also represented as 32 32-bit generic registers (W).

Reply:
Ok.. seems good.